workflows/pipe-common/pipeline/hpc/gridengine.py

```diff
@@ -484,3 +546,226 @@ def validate(self, jobs):
+
+    def _get_host_state(self, host):
+        try:
+            for line in self.cmd_executor.execute_to_lines(SlurmGridEngine._SHOW_EXECUTION_HOST % host):
+                if "NodeName" in line:
+                    return self._parse_dict(line).get("State", "UNKNOWN")
+        except ExecutionError as e:
+            Logger.warn("Problems with getting host '%s' info: %s" % (host, e))
+            return "UNKNOWN"
+
+    def _parse_jobs(self, scontrol_jobs_output):
+        jobs = []
+        jobs_des_lines = [line for line in scontrol_jobs_output.splitlines() if "JobId=" in line]
+        for job_desc in jobs_des_lines:
+            job_dict = self._parse_dict(job_desc)
+            resources = self._parse_dict(job_dict.get("TRES"), line_sep=",")
+            general_resources = self._parse_dict(job_dict.get("GRES"), line_sep=",")
+            num_node = int(re.match("(\\d+)-?.*", job_dict.get("NumNodes", "1")).group(1))
+            # Splitting one job on 'num_node' ephemeral jobs. The idea is to instruct autoscaler that we need to spread
+            # this job to `num_node` nodes and provide portion of resources
+            # TODO maybe there is another way to achieve that?
+            for node_idx in range(num_node):
+                jobs.append(
+                    GridEngineJob(
+                        id=job_dict.get("JobId") + "_" + str(node_idx),
+                        root_id=job_dict.get("JobId"),
+                        name=job_dict.get("JobName"),
+                        user=self._parse_user(job_dict.get("UserId")),
+                        state=GridEngineJobState.from_letter_code(job_dict.get("JobState")),
+                        datetime=self._parse_date(
+                            job_dict.get("StartTime") if job_dict.get("StartTime") != "Unknown" else job_dict.get("SubmitTime")),
+                        hosts=self._parse_nodelist(job_dict.get("NodeList")),
+                        cpu=int(job_dict.get("NumCPUs", "1")) // num_node,
+                        gpu=0 if "gpu" not in general_resources else int(general_resources.get("gpu")) // num_node,
+                        mem=self._parse_mem(self._find_memory_value(job_dict, resources))
+                    )
+                )
+        return jobs
+
+    def _parse_date(self, date):
+        return datetime.strptime(date, SlurmGridEngine._SCONTROL_DATETIME_FORMAT)
+
+    def _parse_dict(self, text, line_sep=" ", value_sep="="):
+        if not text:
+            return {}
+        return {
+            key_value[0]: key_value[1] if len(key_value) == 2 else "" for key_value in
+            [
+                entry.split(value_sep, 1) for entry in text.split(line_sep)
+            ]
+        }
+
+    def _find_memory_value(self, job_dict, resource_dict):
+        if "MinMemoryNode" in job_dict:
+            return job_dict.get("MinMemoryNode")
+        elif "mem" in resource_dict:
+            resource_dict.get("mem")
```

Comment:
Missing return?

```suggestion
            return resource_dict.get("mem")
```

As long as we have such complex parsing, let's add tests to cover all corner cases which we use in `self._parse_jobs`.